doc/source/reference/concepts.md

```diff
@@ -0,0 +1,98 @@
+# Seldon Core Concepts
+
+This page is a work in progress to provide concepts related to Seldon Core.
+
+This work is ongoing and we welcome feedback
+
+## Workflow
+#### Conceptual overview of workflows
+A workflow groups the components of a machine learning system into a logical pipeline. Sets of related components are grouped into graphs. The workflow contains the configuration of the components and the definitions of the inputs and outputs of the system, and of each component.
+
+## Component
+#### Conceptual overview of components
+A component is one of:
+-   Model
+    
+-   Router
+    
+-   Combiner
+    
+-   Transformer
+    
+-   Output Transformer
+
+A program that performs one step in the workflow.
+
+## Machine Learning Deployment
+#### Conceptual overview of machine learning deployments
+A machine learning deployment refers to a predictive model in the Seldon ecosystem of a type associated with Seldon (Seldon Deployments) or Kubeflow (Inference Services).
+
+## Model
+#### Conceptual overview of models
+A component within a machine learning deployment which holds the representation of learned information from the training data.
+
+## Language Wrapper
+#### Conceptual overview of language wrappers
+Enables cross language and/or runtime interoperability of a model with a particular programming language.
+
+## Pre-packaged Inference Server
+#### Conceptual overview of pre-packaged inference servers
+A pre-packaged inference server is one of:
```

Comment:
We can also clarify that pre-packaged inference servers come built into Seldon Core, to allow users to go easily from artifact (i.e. serialised model) to ML deployment regardless of toolkit. 

We can also link to this docs page for users looking for instructions on how to create their own "pre-packaged" inference servers: 

- https://docs.seldon.io/projects/seldon-core/en/latest/servers/overview.html
- https://docs.seldon.io/projects/seldon-core/en/latest/servers/custom.html